core/chaincode/lifecycle/cache_test.go

```diff
@@ -1143,6 +1145,113 @@ var _ = Describe("Cache", func() {
 					Expect(installedCC.References["channel-id"]).To(HaveLen(1))
 				})
 			})
+
+			Context("when a chaincode definition is updated by an approve event with an empty package ID for existing chaincode name and sequence number", func() {
+
+				BeforeEach(func() {
```

Comment:
This BeforeEach has a considerable amount of identical code as the one below - could you please unify the contexts, have a common BeforeEach and highlight the differences in two nested contexts?